### PR TITLE
AB#52 Implement Walking Intent

### PR DESCRIPTION
**Pull request by StuckiSimon**
Basic implementation for MoveIntent.

Uses `Immer` for handling state mutations nicely.
Define basic test for MoveIntent.
Define reusable Test StateBuilder.

### CODE REVIEW

File: game-backend/src/game-engine/intent/MoveIntent.test.ts [hunk 1]
```diff
@@ -1,26 +1,63 @@
...
-        objectIds: ['unit1'],
-      },
+      StateBuilder.buildState()
+        .addObject({ id: 'unit1', x: 0, y: 0 })
+        .build(),
       {
         unitId: 'unit1',
       }
     );
     expect(newState.objectMap.unit1.position.y).toBe(1);
```
Comment (michaelschufi) [inline game-backend/src/game-engine/intent/MoveIntent.test.ts]: Where do we want the origin of our coordinate system to be in? in the center, top-left, bottom-left?
Imo it should be in the center or bottom-left. Having a "move down" intent increasing a axis value feels really weird.
What do you guys think? @sir-marc @flipmess @ferinator @mailaenderli @StuckiSimon

Comment (StuckiSimon) [reply game-backend/src/game-engine/intent/MoveIntent.test.ts]: For rendering in the frontend the origin will be top left (`canvas`). Therefore I'd like to use the same reference point in the backend as well.

Comment (michaelschufi) [reply game-backend/src/game-engine/intent/MoveIntent.test.ts]: That's a good point. However, I still find the Cartesian coordinate system way more intuitive. The canvas is also not doing any calculations with the coordinates. Flipping the y-axis should happen in the frontend and not in the backend imo. 

But maybe it's just me, who feels so put off by a y-axis pointing to the right of the x-axis.

File: .eslintrc [hunk 2]
```diff
@@ -26,6 +26,7 @@
     "no-plusplus": "off",
     "no-useless-constructor": "off",
     "no-empty-function": "off",
+    "no-param-reassign": ["error", { "props": true, "ignorePropertyModificationsFor": ["draft"] }],
```
Comment (michaelschufi) [inline .eslintrc:29]: I assume this is for Immer?

File: game-backend/src/game-engine/intent/MoveIntent.test.ts [hunk 3]
```diff
@@ -1,26 +1,63 @@
...
+
+    const newState = intent.execute(
+      StateBuilder.buildState()
+        .addObject({ id: 'unit1', x: 0, y: ARENA_HEIGHT })
+        .build(),
+      {
+        unitId: 'unit1',
+      }
+    );
+    expect(newState.objectMap.unit1.position.y).toBe(ARENA_HEIGHT);
```
Comment (michaelschufi) [inline game-backend/src/game-engine/intent/MoveIntent.test.ts]: We should also test the x-coordinate? We could put it in the same test.

**Review by michaelschufi — CHANGES_REQUESTED** [review]
Well done. I like the StateBuilder ;)
We could add tests for the x-coordinate and we should discuss the y-axis problem.

File: .eslintrc (hunk 2 above)
Comment (StuckiSimon) [reply .eslintrc:29]: Yes, it is.

File: game-backend/src/game-engine/intent/MoveIntent.test.ts (hunk 3 above)
Comment (StuckiSimon) [reply game-backend/src/game-engine/intent/MoveIntent.test.ts]: I've added a new test :v:

Now all acceptance criterias are tested using automated unit tests 🌮

File: game-backend/src/game-engine/intent/MoveIntent.test.ts (hunk 1 above)
Comment (StuckiSimon) [reply game-backend/src/game-engine/intent/MoveIntent.test.ts]: As discussed with the team, we'll use the upper left as the origin of the coordinate system.

**Review by michaelschufi — APPROVED** [review]
Thanks for the tests.

**Pull request merged**